meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/util/ZkMetaClientUtil.java

```diff
@@ -0,0 +1,201 @@
+   * zk transactional support.
+   * @param ops
+   * @return
+   */
+  public static List<Op> metaClientOpToZk(Iterable<org.apache.helix.metaclient.api.Op> ops) {
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.CREATE, op -> {
+      try {
+        CreateMode mode = convertMetaClientMode(((org.apache.helix.metaclient.api.Op.Create) op).getEntryMode());
+        return Op.create(op.getPath(), ((org.apache.helix.metaclient.api.Op.Create) op).getData(), DEFAULT_ACL, mode);
+      } catch (KeeperException e) {
+        throw translateZkExceptionToMetaclientException(ZkException.create(e));
+      }
+    });
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.DELETE,
+        op -> Op.delete(op.getPath(), ((org.apache.helix.metaclient.api.Op.Delete) op).getVersion()));
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.SET,
+        op -> Op.setData(op.getPath(),
+            ((org.apache.helix.metaclient.api.Op.Set) op).getData(), ((org.apache.helix.metaclient.api.Op.Set) op).getVersion()));
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.CHECK,
+        op -> Op.check(op.getPath(), ((org.apache.helix.metaclient.api.Op.Check) op).getVersion()));
+
+    List<Op> zkOps = new ArrayList<>();
+    for (org.apache.helix.metaclient.api.Op op : ops) {
+      Function<org.apache.helix.metaclient.api.Op, Op> function = getOpMap().get(op.getType());
+      if (function != null) {
+        zkOps.add(function.apply(op));
+      } else {
+        throw new IllegalArgumentException("Op type " + op.getType().name() + "is not supported.");
+      }
+    }
+    return zkOps;
+  }
+
+  private static final class OpmapHolder {
+    static final Map<org.apache.helix.metaclient.api.Op.Type, Function<org.apache.helix.metaclient.api.Op, Op>> OPMAP =
+        new EnumMap<>(org.apache.helix.metaclient.api.Op.Type.class);
+  }
+
+  private static Map<org.apache.helix.metaclient.api.Op.Type, Function<org.apache.helix.metaclient.api.Op, Op>> getOpMap() {
+    return OpmapHolder.OPMAP;
+  }
+
+  private static CreateMode convertMetaClientMode(MetaClientInterface.EntryMode entryMode) throws KeeperException {
+    if (entryMode == MetaClientInterface.EntryMode.PERSISTENT) {
+      return CreateMode.fromFlag(0);
+    }
+    if (entryMode == MetaClientInterface.EntryMode.EPHEMERAL) {
+      return CreateMode.fromFlag(1);
+    }
+    if (entryMode == MetaClientInterface.EntryMode.CONTAINER) {
+      return CreateMode.fromFlag(4);
+    }
+    throw new IllegalArgumentException(entryMode.name() + " is not a supported EntryMode.");
+  }
+
+  /**
+   * Helper function for transactionOP. Converts the result from calling zk transactional support into
+   * metaclient OpResults.
+   * @param zkResult
+   * @return
+   */
+  public static List<OpResult> zkOpResultToMetaClient(List<org.apache.zookeeper.OpResult> zkResult) {
+    getOpResultMap().put(org.apache.zookeeper.OpResult.CreateResult.class, opResult -> {
```

Comment:
This doesn't look like lazy loading. Each time the method invoked, we still go through all map.put() process.